src/Components/CutPlaneMenu.jsx

```diff
@@ -114,10 +121,147 @@ export default function CutPlaneMenu() {
           },
         }}
       >
-        <MenuItem onClick={() => createPlane('x')} selected={cutPlaneDirection === 'x'}> X</MenuItem>
-        <MenuItem onClick={() => createPlane('y')} selected={cutPlaneDirection === 'y'}>Y</MenuItem>
-        <MenuItem onClick={() => createPlane('z')} selected={cutPlaneDirection === 'z'}>Z</MenuItem>
+        <MenuItem onClick={() => createPlane('x', 0)} selected={cutPlaneDirection === 'x'}> X</MenuItem>
+        <MenuItem onClick={() => createPlane('y', 0)} selected={cutPlaneDirection === 'y'}>Y</MenuItem>
+        <MenuItem onClick={() => createPlane('z', 0)} selected={cutPlaneDirection === 'z'}>Z</MenuItem>
       </Menu>
     </div>
   )
 }
+
+
+/* eslint-disable no-magic-numbers */
+/**
+ * getSelectionAxisFromBoundingBox is the helper method for the cutplane logic
+ *
+ * @param {object} boundingBox bouding box
+ * @return {object}
+ */
+export function getSelectionAxisFromBoundingBox(boundingBox) {
+  return {
+    x: {
+      size: Math.abs( boundingBox.max.x - boundingBox.min.x),
+      center: (boundingBox.max.x + boundingBox.min.x) / 2,
+    },
+    y: {
+      size: Math.abs( boundingBox.max.y - boundingBox.min.y),
+      center: (boundingBox.max.y + boundingBox.min.y) / 2,
+    },
+    z: {
+      size: Math.abs(boundingBox.max.z - boundingBox.min.z ),
+      center: (boundingBox.max.z + boundingBox.min.z) / 2,
+    },
+  }
+}
+
+
+/**
+ * getModelCenter return the center of the model based on bounding box
+ *
+ * @param {object} ifcModel bouding box
+ * @return {object} centerCoordinates
+ */
+export function getModelCenter(ifcModel) {
+  return new Vector3(
```

Comment:
as noted in discussion, i think you can pass in bb and then use the one-liner from three.js to do the calculation